src/account.c

```diff
@@ -1241,10 +1241,58 @@ int account_debug(struct re_printf *pf, const struct account *acc)
 		}
 		err |= re_hprintf(pf, "\n");
 	}
-	err |= re_hprintf(pf, " call_transfer:         %s\n",
+	err |= re_hprintf(pf, " call_transfer:%s\n",
 			  account_call_transfer(acc));
-	err |= re_hprintf(pf, " extra:         %s\n",
+	err |= re_hprintf(pf, " extra:        %s\n",
 			  acc->extra ? acc->extra : "none");
 
 	return err;
 }
+
+
+/**
+ * Print the account information in JSON
+ *
+ * @param od  Account dict
+ * @param odcfg  Configuration dict
+ * @param acc User-Agent account
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int account_json_api(struct odict *od, struct odict *odcfg, const struct account *acc)
+{
+	int err = 0;
+
+	if (!acc)
+		return 0;
+
+	/* account */
+	err |= odict_entry_add(od, "aor", ODICT_STRING, acc->aor);
+	err |= odict_entry_add(od, "display_name", ODICT_STRING, acc->dispname);
+
+	/* config */
+	err |= odict_entry_add(odcfg, "sip_nat", ODICT_STRING, account_sipnat(acc));
+
+	struct odict *obn = NULL;
+	err |= odict_alloc(&obn, 8);
+	for (size_t i=0; i<ARRAY_SIZE(acc->outboundv); i++) {
+		if (acc->outboundv[i]) {
+			err |= odict_entry_add(obn, "outbound", ODICT_STRING, acc->outboundv[i]);
+		}
+	}
+	err |= odict_entry_add(odcfg, "sip_nat_outbound", ODICT_ARRAY, obn);
+
+	const char *stunhost = account_stun_host(acc) ? account_stun_host(acc) : "";
+	err |= odict_entry_add(odcfg, "stun_host", ODICT_STRING, stunhost);
+	err |= odict_entry_add(odcfg, "stun_port", ODICT_INT, account_stun_port(acc));
+	err |= odict_entry_add(odcfg, "stun_user", ODICT_STRING, account_stun_user(acc));
+	const char *stunpwd = account_stun_pass(acc) != NULL ? "***" : "";
```

Comment:
perhaps we should consider dropping printing of stun password ?

Reply:
Yes, I agree, did not think it's a good idea to send it out in the api, so obscuring it to show it's there, is maybe not needed.